services/app-api/src/emailer/templateHelpers.ts

```diff
@@ -78,36 +78,63 @@ const generateCMSReviewerEmails = (
     config: EmailConfiguration,
     pkg: LockedHealthPlanFormDataType | UnlockedHealthPlanFormDataType,
     stateAnalystsEmails: StateAnalystsEmails
-): string[] => {
-    //Combine CMS emails along with State specific analyst emails.
-    const cmsReviewSharedEmails = [
-        ...config.cmsReviewSharedEmails,
-        ...stateAnalystsEmails,
-    ]
+): string[] | Error => {
+    // OACT email, receives only CONTRACT_AND_RATES submission, excludes CHIP and state of PR
+    const oactEmails: string[] = config.ratesReviewSharedEmails
+
+    // DMCP and OACT share an inbox that receives all submissions, excludes CHIP and state of PR and should be included
+    // in cmsReviewSharedEmails.
+    // cmsRateHelpEmailAddress is also set to this shared DMCP and OACT inbox. We are setting this const in order to
+    // filter it out for CHIP and State of PR submissions because we cannot store the actual email in the API. This
+    // will eventually be fixed when moving to storing emails in our DB.
+    const dmcpOACTSharedEmail: string = config.cmsRateHelpEmailAddress
+
+    // CHIP programs and state of PR submission does not include OACT and DMCP emails.
+    const filterChipAndPRSubmission = (reviewers: string[]) =>
+        reviewers.filter(
+            (email) =>
+                email !== dmcpOACTSharedEmail && !oactEmails.includes(email)
+        )
 
     const programIDs = findAllPackageProgramIds(pkg)
 
-    //chipReviewerEmails does not include OACT and DMCP emails
-    const chipReviewerEmails = cmsReviewSharedEmails.filter(
-        (email) => email !== config.cmsRateHelpEmailAddress
-    )
-    const contractAndRateReviewerEmails = [
-        ...cmsReviewSharedEmails,
-        ...config.ratesReviewSharedEmails,
-    ]
-
-    if (
-        pkg.submissionType === 'CONTRACT_AND_RATES' &&
-        pkg.stateCode !== 'PR' &&
-        !includesChipPrograms(programIDs)
-    ) {
-        return pruneDuplicateEmails(contractAndRateReviewerEmails)
-        //
-    } else if (includesChipPrograms(programIDs)) {
-        return pruneDuplicateEmails(chipReviewerEmails)
-    }
+    // dmcpEmail is added in both CONTRACT_ONLY and CONTRACT_AND_RATES, in case it wasn't included in cmsReviewSharedEmails.
```

Comment:
I appreciate the named variables, it helps a lot. I would also like to see a set of table tests  for `templateHelpers.ts` specifically the `generateCMSReviewerEmails` function. I know we should be also testing this through the templates ...  but whenever I see a utility function manipulating an array in several ways I see an opportunity to also quickly unit test and hit each logic branch (e.g. submission type with chip programs, submission type without chip, etc).

Reply:
Added tests for `generateCMSReviewerEmails` in a new `templateHelpers.test.ts`. Tests include...
- `'Contract only submission'`
- `'Contract and rates submission'`
- `'Submission with CHIP program specified for contract certification'`
- `'Submission with CHIP program specified for rate certification'`
- `'Puerto Rico submission'`
- `'Error result.'`